src/transformers/visitors/orchestrationInternalFunctionCallVisitor.ts

```diff
@@ -507,10 +525,64 @@ FunctionCall: {
         state.circuitImport.push('true');
         else
         state.circuitImport.push('false');
-        const newNode = buildNode('InternalFunctionCall', {
+      let newNode;
+        if(parent.nodeType === 'VariableDeclarationStatement') {
+         
+          const decNode = buildNode('VariableDeclarationStatement')
+          decNode.declarations.push(functionReferncedNode.node.returnParameters.parameters[0]._newASTPointer);
+          decNode.interactsWithSecret = true;
+          decNode.declarations[0].declarationType = 'state';
+          decNode.declarations[0].isAccessed = true;
+          decNode.declarations[0].interactsWithSecret = true;
+          state.decNode = decNode;
+          const returnPara = functionReferncedNode.node.returnParameters.parameters[0].name;
+          let includeExpressionNode = false;
+          // this functions checks if the parent node interact with secret in the calling function or not
+          callingfnDefIndicators[parent.declarations[0].id].interactsWith.forEach( node => {
+            if(node.key != 'arguments' && node.interactsWithSecret)
+            includeExpressionNode = true;
+           })
+           includeExpressionNode ? 
+          functionReferncedNode.node.body.statements.forEach(exp => {
+            // If the return para interacts with public only in the internal function but with secret in calling function we need this expression in calling function
+          if(exp?.expression.leftHandSide?.name === returnPara && !exp.expression.leftHandSide.interactsWithSecret){
+            const newName =returnPara+'_1';
+            const innerNode = buildNode('VariableDeclaration', {
+              name: newName,
+              isAccessed: true,
+              isSecret: false,
+              interactsWithSecret: true, // setting interact with secret true, as now it interacts with secret in the calling function
+            });
+            const initNode = buildNode('BinaryOperation', {
+              leftExpression: exp._newASTPointer.expression.rightHandSide.leftExpression,
+              operator: exp._newASTPointer.expression.rightHandSide.operator,
+              rightExpression: exp._newASTPointer.expression.rightHandSide.rightExpression,
+            });
+            state.expNode = buildNode('VariableDeclarationStatement',{
```

Comment:
Is state.expNode not replaced for every exp in functionReferncedNode.node.body.statements?